### PR TITLE
net/ntopng: add https support

### LINKED ISSUE #952: NTOPNG plugin needs way to set https
You currently cannot use the ntopng plugin unless you 
1. enable ssh access to opnsense
2. enable https in the console config (/usr/local/etc/ntopng.conf)
3. remove the non-http config option
4. supply a certificate for https (/usr/local/share/ntopng/httpdocs/ssl)
5. restart ntopng

Would be nice if we could do it from the plugin interface.

### PR DESCRIPTION
**Pull request by mimugmail**
Closes #952

### CODE REVIEW

File: net/ntopng/Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 PLUGIN_NAME=		ntopng
 PLUGIN_VERSION=		1.0
-PLUGIN_REVISION=    1
+PLUGIN_REVISION=    2
```
Comment (fichtner) [inline net/ntopng/Makefile]: prefer 1.1 and removing port revision -- it's a feature release

File: net/ntopng/src/opnsense/service/templates/OPNsense/Ntopng/ntopng.conf [hunk 2]
```diff
@@ -6,6 +6,9 @@
 {%   if helpers.exists('OPNsense.ntopng.general.httpport') and OPNsense.ntopng.general.httpport != '' %}
 -w={{ OPNsense.ntopng.general.httpport }}
 {%   endif %}
+{%   if helpers.exists('OPNsense.ntopng.general.httpport') and OPNsense.ntopng.general.httpport != '' %}
```
Comment (fichtner) [inline net/ntopng/src/opnsense/service/templates/OPNsense/Ntopng/ntopng.conf]:
```suggestion
{%   if helpers.exists('OPNsense.ntopng.general.httpsport') and OPNsense.ntopng.general.httpsport != '' %}
```

**Comment by fichtner** [comment]
@mimugmail what's the logic to enable https: set port or certificate or both? it's only loosely enforced and setting a certificate without a port is possible leading to confusion. may be better to require the port, set a default and trigger via SSL certificate?

**Comment by mimugmail** [comment]
@fichtner the service doesn't start when httpsport not empty but certificate is none

**Comment by fichtner** [comment]
sounds like we need a constraint to cover this then: settings https port requires a certficate != none?

**Comment by mimugmail** [comment]
now it looks better :)

**Comment by fichtner** [comment]
ok, this requires unsetting both port and cert to deactivate, right?

**Comment by mimugmail** [comment]
correct, both must be set or empty

**Comment by fichtner** [comment]
Merged, thanks! This is good, although @fabianfrz had a constraint working that required a different setting to be set as well which is a better fix in the long-term (drop cert, leave port set), but that also requires templating changes. (Fabian, please correct me if I'm wrong about the nginx constraint you had).

**Comment by fabianfrz** [comment]
@fichtner the constraint for nginx does effectively this:

grafik

As you can see the SNI based one can be none, while the upstram is required when route with upstream is chosen.

For booleans or simple empty checks, there is https://github.com/opnsense/core/blob/master/src/opnsense/mvc/app/models/OPNsense/Base/Constraints/DependConstraint.php

**Comment by fichtner** [comment]
@fabianfrz yep, can you add a core PR?

**Comment by mimugmail** [comment]
Nice approach, but in this specific case there is no benefit, correct?
Cause why set https port when no certificate or why set certificate when there's no https enabled.

**Comment by fichtner** [comment]
there is the benefit of slightly better UX, you only need one field to disable HTTPS

**Comment by fabianfrz** [comment]
@fichtner I removed the prefix and it is now in opnsense/core#2845

**Pull request merged**